### PR TITLE
Add diagnostic help for inline asm operand constraint 'H' 

### PR DESCRIPTION
**Pull request by mahesh-attarde**
Using inline asm operand constraint 'H' with constants does not reflect helpful error. 
With addition of this message operands those are not offsetable, will report exact root cause.
Originally https://reviews.llvm.org/D35890

### CODE REVIEW

File: llvm/test/CodeGen/X86/inline-asm-modifier-error.ll [hunk 1]
```diff
@@ -0,0 +1,9 @@
+; RUN: not llc -march x86 < %s 2> %t
+; RUN: FileCheck %s < %t
```
Comment (arsenm) [inline llvm/test/CodeGen/X86/inline-asm-modifier-error.ll:2]: Don't use a temporary file
```suggestion
; RUN: not llc -march x86 < %s 2>&1 | FileCheck %s
```

File: llvm/test/CodeGen/X86/inline-asm-modifier-error.ll [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+; CHECK: error: invalid operand in inline asm: 'movl $0, ${1:H}' 'H' modifier used on an operand that is a non-offsetable memory reference.
+define void @H() local_unnamed_addr #0 {
```
Comment (arsenm) [inline llvm/test/CodeGen/X86/inline-asm-modifier-error.ll:5]:
```suggestion
define void @H() {
```

**Review by arsenm — COMMENTED** [review]
I think it would be a better API to migrate the print from return bool to Error instead of adding a string out argument

**Comment by mahesh-attarde** [comment]
> I think it would be a better API to migrate the print from return bool to Error instead of adding a string out argument

This is tricky. AsmPrinter::PrintAsmOperand is overridden in each target. 'H' can be slightly different  for each target. So when we return to caller of PrintAsmOperand ( generalization) we may need separate special message as per Target. To support it, i added std::string parameter.  Is that okay?

**Comment by arsenm** [comment]
> > I think it would be a better API to migrate the print from return bool to Error instead of adding a string out argument
> 
> This is tricky. AsmPrinter::PrintAsmOperand is overridden in each target. 'H' can be slightly different for each target. So when we return to caller of PrintAsmOperand ( generalization) we may need separate special message as per Target. To support it, i added std::string parameter. Is that okay?

Error carries along with it any error string you want

**Comment by mahesh-attarde** [comment]
https://github.com/llvm/llvm-project/pull/88248
working patch from other pull request. something went wrong with git files.

**Pull request closed without merging**